src/plugin.js

```diff
@@ -9,6 +9,26 @@ export default function (Vue) {
         from: 'experiment',
         default: () => ({})
       }
+    },
+
+    magpie: config,
+
+    /**
+     * Register all socket events
+     */
+    mounted() {
+      if (this.$options.socket) {
+        this.$exp.setUpSubscriptions(this.$options.socket);
```

Comment:
I think I felt a bit lost why it is `this.$exp.setUpSubscriptions` instead of something like `this.$exp.socket.setUpSubscriptions`. Is it because with

```
    inject: {
      $exp: {
        from: 'experiment',
        default: () => ({})
      }
    }
```

everything from the `Experiment` component is injected in the outmost scope?

Reply:
But `setUpSubscription` is still under `Socket` instead of being a part of the `methods()` of the `Experiment` component, or?...

Reply:
Yes, you're right. This should be socket.setUpSubscriptions